creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -77,14 +134,25 @@ class SnapbackSM {
       creatorNodeEndpoint: this.endpoint
     })
 
-    // Config to determine if reconfig is enabled
-    this.snapbackReconfigEnabled = this.nodeConfig.get('snapbackReconfigEnabled')
-
     // The interval when SnapbackSM is fired for state machine jobs
     this.snapbackJobInterval = this.nodeConfig.get('snapbackJobInterval') // ms
 
     // Mapping of Content Node endpoint to its service provider ID
     this.endpointToSPIdMap = {}
+
+    // We do not want to eagerly cycle off the primary when issuing reconfigs if necessary, as the primary may
+    // have data that the secondaries lack. In this case, wait until `this.moduloBase` iterations for the
+    // primary to potentially become healthy again. This set is used to track visited primaries.
```

Comment:
is there a way to abstract this logic inside peerSetManager?

Reply:
we can move any node health related code to peerSetManager to simplify code in this file

Reply:
also merge any similar logic. we perform a simple health check on primary and can abstract existing logic to handle simple health checks too